### PR TITLE
Add more description about CPD vs ICP filters

### PR DESCRIPTION
**Pull request by sunapi386**
I found the documentation on ICP vs CPD a little too similar, and wasn't sure when to use which. I added some more description after reading through the paper here http://graphics.stanford.edu/courses/cs468-07-winter/Papers/nips2006_0613.pdf.

### CODE REVIEW

**Comment by sunapi386** [comment]
@gadomski I made the changes, squashed commits.

File: doc/stages/filters.cpd.rst [hunk 1]
```diff
@@ -22,6 +22,23 @@ the "moving" points after the calculated transformation has been applied, one
 point view per input.  Any additional information about the cpd registration,
 e.g. the rigid transformation matrix, will be placed in the stage's metadata.
 
+When to use CPD vs ICP
```
Comment (gadomski) [inline doc/stages/filters.cpd.rst]: Got this warning:

```
/Users/rdcrlpjg/Repos/PDAL/doc/stages/filters.cpd.rst:26: WARNING: Title underline too short.
When to use CPD vs ICP
---------
```

**Comment by sunapi386** [comment]
Pedantic is good. I guess that issue didn't show up the first time.

**Comment by gadomski** [comment]
No problem at all, I think I missed it in my first review. 👍 looks good to me!

**Pull request merged**